### PR TITLE
Gig template

### PR DESCRIPTION
**Pull request by aaronoppenheimer**
Added a bunch of fields to the gig edit form, including the "send update" options. Moved the code to send updates into the view handling for both creating and editing gigs, removing the signal that used to do it on gig save. Updated all tests to use form/view instead of just saving gig objects.

### CODE REVIEW

File: gig/tests.py [hunk 1]
```diff
@@ -133,49 +193,21 @@ def test_new_gig_email(self):
...
-    def test_gig_time_long_end(self):
-        date = timezone.datetime(2100, 1, 2, 12, tzinfo=pytz_timezone(self.band.timezone))
-        self.assoc_joe_and_create_gig(start_date=date, set_date=None, end_date=date + timedelta(days=1))
-        self.assertIn('Call Time: 01/02/2100 noon (Sat)\nEnd Time: 01/03/2100 noon (Sun)\nContact', mail.outbox[0].body)
-
-    def test_gig_time_long_set_end(self):
-        date = timezone.datetime(2100, 1, 2, 12, tzinfo=pytz_timezone(self.band.timezone))
-        self.assoc_joe_and_create_gig(start_date=date, set_date=date + timedelta(days=1), end_date=date+timedelta(days=1, hours=1))
-        self.assertIn('Call Time: 01/02/2100 noon (Sat)\nSet Time: 01/03/2100 noon (Sun)\nEnd Time: 01/03/2100 1 p.m. (Sun)\nContact', mail.outbox[0].body)
-
```
Comment (rschroll) [inline gig/tests.py]: Why are these tests removed?  Proper display of this information relies on some rather hairy template code that really needs some testing.

Comment (aaronoppenheimer) [reply gig/tests.py]: It was inadvertent, then I didn't look carefully when I checked it in. Have reinstated them!

File: lib/caldav.py [hunk 2]
```diff
@@ -52,11 +52,9 @@ def _make_description(event):
         """ description is the details, plus the setlist """
         x = ''
         if e.details:
-            # todo - need to do this replace?
-            x = '{0}'.format(e.details.replace('\r\n', '\\n'))
+            x = e.details
         if e.setlist:
-            x = '{0}\\n\\n{1}'.format(x, e.setlist.replace(
-                '\r\n', '\\n'))  # todo - need to do this replace?
+            x = f'{x}\\n\\n{e.setlist}'
```
[the remark this answers is not in the document]
Comment (aaronoppenheimer) [reply lib/caldav.py]: Ha - should clearly be 'event' which means it wasn't tested properly. Will clean up.

File: gig/helpers.py [hunk 3]
```diff
@@ -155,3 +157,8 @@ def send_snooze_reminders():
                                    gig__date__gt=now)
     send_emails_from_plans(unsnooze, 'email/gig_reminder.md')
     unsnooze.update(snooze_until=None)
+
+def notify_new_gig(gig, created):
+    # This will have the side effect of creating plans for all members
```
[the remark this answers is not in the document]
Comment (aaronoppenheimer) [reply gig/helpers.py]: nope!

File: gig/views.py [hunk 4]
```diff
@@ -46,6 +52,16 @@ def get_context_data(self, **kwargs):
         context['the_band'] = Band.objects.get(id=self.kwargs['bk'])
         return context
 
+    # def get_initial(self):
+    #     initial = super(CreateView, self).get_initial()
+    #     initial['contact'] = Band.objects.get(id=self.kwargs['bk']).confirmed_members
+    #     return initial
```
[the remark this answers is not in the document]
Comment (aaronoppenheimer) [reply gig/views.py]: agreed

**Comment by aaronoppenheimer** [comment]
Updated with requested changes

**Pull request merged**